administrator/components/com_fields/helpers/fields.php

```diff
@@ -170,8 +173,23 @@ public static function getFields ($context, $item = null, $prepareValue = false)
 					{
 						try
 						{
+							/*
+							 *On before field render
+							 *Event allow plugins to modfify the output of the field before it is rendered
+							 */
+                            $dispatcher = JEventDispatcher::getInstance();
+							if($field->value)
+								$dispatcher->trigger('onFieldBeforeRender',array($context,'mustache',&$field));
```

Comment:
I put the triggers in the render method because at that point we have all the information we need. unfortunately for developers like me who have an extension that is a plugin and not a component or a template we will not be able to change the layout of the rendered custom field with a layout override. And in most cases even if  we  could get to the layout it will be too late by then as the information  we would need will not be present at that stage Or the render data will by in a format to late for us to change. 

So a real life example will be inline editing when we need to make the values of the custom fields clickable editable regions when the user is logged into the frontend of they website and want to update a custom field.

However, if you have an better ideas I am open to another way of doing it.